### PR TITLE
Properly handle too-large messages

### PR DESCRIPTION
**Pull request by michaelkirk**
PTAL @charlesmchen

### CODE REVIEW

File: SignalServiceKit/src/Messages/OWSMessageReceiver.m [hunk 1]
```diff
@@ -333,6 +333,28 @@ + (void)syncRegisterDatabaseExtension:(YapDatabase *)database
...
+            return (@{
+                @"message_size" : @([OWSAnalytics orderOfMagnitudeOf:(long)envelope.serializedSize]),
+            });
+        });
+        return;
+    }
+
+    // Take note of any messages larger than we expect, but still process them.
+    // This likely indicates a misbehaving sending client.
+    NSUInteger kLargeEnvelopeWarningByteCount = 25 * 1024;
```
Comment (michaelkirk) [inline SignalServiceKit/src/Messages/OWSMessageReceiver.m]: I picked this randomly. We could probably be even more conservative.

File: SignalServiceKit/src/Messages/OWSMessageReceiver.m [hunk 2]
```diff
@@ -333,6 +333,28 @@ + (void)syncRegisterDatabaseExtension:(YapDatabase *)database
 
 - (void)handleReceivedEnvelope:(OWSSignalServiceProtosEnvelope *)envelope
 {
+    // Drop any too-large messages on the floor. Well behaving clients should never send them.
+    NSUInteger kMaxEnvelopeByteCount = 250 * 1024;
+    if (envelope.serializedSize > kMaxEnvelopeByteCount) {
+        OWSProdErrorWParams(@"message_receiver_error_oversize_message", ^{
+            return (@{
+                @"message_size" : @([OWSAnalytics orderOfMagnitudeOf:(long)envelope.serializedSize]),
```
Comment (charlesmchen) [inline SignalServiceKit/src/Messages/OWSMessageReceiver.m]: Per our privacy review, please just use `OWSProdError()` without the params.

File: SignalServiceKit/src/Messages/OWSMessageReceiver.m [hunk 3]
```diff
@@ -333,6 +333,28 @@ + (void)syncRegisterDatabaseExtension:(YapDatabase *)database
...
+            });
+        });
+        return;
+    }
+
+    // Take note of any messages larger than we expect, but still process them.
+    // This likely indicates a misbehaving sending client.
+    NSUInteger kLargeEnvelopeWarningByteCount = 25 * 1024;
+    if (envelope.serializedSize > kLargeEnvelopeWarningByteCount) {
+        OWSProdErrorWParams(@"message_receiver_error_large_message", ^{
```
Comment (charlesmchen) [inline SignalServiceKit/src/Messages/OWSMessageReceiver.m]: Ditto.

File: SignalServiceKit/src/Messages/OWSMessageReceiver.m (hunk 1 above)
Comment (charlesmchen) [inline SignalServiceKit/src/Messages/OWSMessageReceiver.m]: I'm guessing that the largest valid envelope should be some small multiple of the largest possible text message.

```
        const NSUInteger kOversizeTextMessageSizeThreshold = 16 * 1024;
```

I'm guessing that a 16k text message would yield an envelope smaller than 25k, but we should verify that?

**Pull request merged**